docs/proposals/29-autoscaling-storage-volumes.md

```diff
@@ -0,0 +1,335 @@
+
+#### Alternative: Scale via controller in `gardenlet`
+**Overview:**
+Scaling in `gardenlet` does not offer advantages which can justify burdening the already large and complex `gardenlet`.
+It also cannot effectively support scaling of shoot PVCs.
+PVC autoscaling is a prime candidate for running as a separate process, as that only imposes the minimal overhead of an
+in-memory PVC object cache and leader election activity.
+
+**Recommendation:**
+It has a major disadvantage, compared to `pvc-autoscaler`.
+
+#### Alternative: Business as usual
+**Overview:**
+The existing approach of statically allocating storage has the average observability volume severely underutilised, yet
+bears a risk of high-load shoots exhausting their allocated storage. 
+
+**Recommendation:**
+The use of `pvc-autoscaler` would result in a substantial economy of storage space. 
+
+**Details:**
+The average observability volume utilisation, observed on a set of typical Gardener shoots, running with the current
+default settings, was:
+- Prometheus average utilisation: 16.6%; &plusmn;1&sigma;: (8.0 - 25.2)
+- Vali average utilisation: 8.3%; &plusmn;1&sigma;: (4.2 - 12.3)
+
+The above aggregations include both shoot and seed volumes.
+
+### Limitations
+#### Reliance on Prometheus
+`pvc-autoscaler` relies on metrics from the seed's cache Prometheus instance. The availability of the otherwise simple,
+and lightly and predictably loaded, autoscaler is thus limited to the availability level of the complex Prometheus, which
+may be subject to load spikes and autoscaling evictions. By itself, this limitation is deemed acceptable, as the trigger
+thresholds, configured for PVC autoscaling, usually afford tolerance to substantial delays.
+
+Using the seed's cache Prometheus instance creates a dependency loop - `pvc-autoscaler` needs `prometheus-cache` in order
+to perform scaling, and `prometheus-cache` needs `pvc-autoscaler` to scale its database volume. That dependency results
+in one mode of persistent failure:
+1. `prometheus-cache` exhausts its storage before `pvc-autoscaler` resizes it
+2. `prometheus-cache` enters a fault loop, due to inability ot expand database
+3. `pvc-autoscaler` enters fault loop, due to driving metrics signal unavailable
+4. At this point, autoscaling seizes for all PVs on the seed
+
+To mitigate the risk of such failure, a larger initial storage request is used for `prometheus-cache` - 6Gi. It offers a
+1Gi headroom over the 5Gi storage retention limit, configured on `prometheus-cache`. Field data suggests 1Gi should
+suffice to accommodate the occasional usage spikes which exceed the data retention limit.
+
+#### Latency
+The pipeline which propagates the volume metrics, driving `pvc-autoscaler`, is as follows:
+1. `kubelet` collects PV state data and publishes resource metrics. Default poll interval: 1 minute
+   ([source](https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/))
```

Comment:
I suspect that the page was updated since I originally referenced it. I replaced the link with the API reference one you provided.

I wonder if, instead of linking to external documentation directly, I should be linking to copies in the Internet Archive.